compiler-plugin/src/main/scala/scalapb/compiler/ProtobufGenerator.scala

```diff
@@ -79,6 +82,7 @@ class ProtobufGenerator(
       }
       .add(s"  case __other => Unrecognized(__other)")
       .add("}")
+      .add(s"sealed trait $RecognizedEnumType extends $name")
```

Comment:
can the declaration come before the various usages? this now appears after.

Reply:
do we get a compilation error when one of the enum values is `Recognized`? If so, we should mutate the name of the enum value to avoid conflict.

Reply:
As the newly added test proves, an enum values `Recognized` doesn't clash with the subtype.
That's because the enum value is a `case object`, while the subtype is a `trait`.

Reply:
See comment here https://github.com/scalapb/ScalaPB/pull/652/files#r324930710, looks good otherwise.